ITR/data/base_providers.py

```diff
@@ -516,47 +577,59 @@ def _compute_missing_historic_ei(self, companies, historic_data):
 
     def _add_projections_to_companies(self, companies: List[ICompanyData], extrapolations: pd.DataFrame):
         for company in companies:
+            if company.company_id not in extrapolations.index.get_level_values(0):
```

Comment:
this check was not here in the original code, and AFAIU, it worked.
why did we need to add an extra conditional branch?

Reply:
If I cannot create a unit test that triggers, I will remove.  Nothing trips is right now.

Reply:
I have removed it because I've removed the code that might cause the extrapolations rows to get out of sync with the list of companies.